jdaviz/configs/specviz/plugins/parsers.py

```diff
@@ -76,17 +80,60 @@ def specviz_spectrum1d_parser(app, data, data_label=None, format=None, show_in_v
             raise ValueError(f"Length of data labels list ({len(data_label)}) is different"
                              f" than length of list of data ({len(data)})")
 
-    with app.data_collection.delay_link_manager_update():
         # these are used to build a combined spectrum with all
         # input spectra included (taken from https://github.com/spacetelescope/
         # dat_pyinthesky/blob/main/jdat_notebooks/MRS_Mstar_analysis/
         # JWST_Mstar_dataAnalysis_analysis.ipynb)
-        wlallorig = []
-        fnuallorig = []
-        dfnuallorig = []
+        wlallorig = []  # to collect wavelengths
+        fnuallorig = []  # fluxes
+        dfnuallorig = []  # and uncertanties (if present)
+
+        for spec in data:
+            for wlind in range(len(spec.spectral_axis)):
+                wlallorig.append(spec.spectral_axis[wlind].value)
+                fnuallorig.append(spec.flux[wlind].value)
+
+                # because some spec in the list might have uncertainties and
+                # others may not, if uncert is None, set to list of NaN. later,
+                # if the concatenated list of uncertanties is all nan (meaning
+                # they were all nan to begin with, or all None), it will be set
+                # to None on the final Spectrum1D
+                if spec.uncertainty[wlind] is not None:
+                    dfnuallorig.append(spec.uncertainty[wlind].array)
+                else:
+                    dfnuallorig.append(np.nan)
+
+    # if the entire uncert. array is Nan and the data is not, model fitting won't
+    # work (more generally, if uncert[i] is nan/inf and flux[i] is not, fitting will
+    # fail, but just deal with the all nan case here since it is straightforward).
+    # set uncerts. to None if they are all nan/inf, and display a warning message.
+    set_nans_to_none = False
+    if isinstance(data, SpectrumList):
+        uncerts = dfnuallorig  # alias these for clarity later on
+        if uncerts is not None:
+            if not np.any(uncerts):
+                uncerts = None
+                set_nans_to_none = True
+    elif data[0].uncertainty is not None:
+        uncerts_finite = np.isfinite(data[0].uncertainty.array)
+        if not np.any(uncerts_finite):
+            if len(data) > 1:
+                raise NotImplementedError('Only supported for 1D data.')
```

Comment:
is there any way from the logic above that `data` could be longer than 1 when it isn't a `SpectrumList`?  I'm a bit confused by what this error statement is trying to catch - no finite uncertainties in the _first_ spectrum and more than one spectrum being added?

Maybe we can add a check for the non-SpectrumList length-of-one assumption before any calls to `data[0]` - just in case that assumption ever breaks in the future?

Reply:
If its not a spectrum list, it will be a spectrum1d but contained in a list for some reason ([Spectrum1D]). I think this is always the case. I might be wrong because theres been a lot of iteration on this PR, but I believe one of the first review comments asked me to put an error here just in case the list didn't contain just one element.

I think that check should be moved up though. If its not a SpectrumList, its a [Spectrum1D]. I should first check that the list has only one element so i can safely index it, then move along.

Reply:
ok done, how does that look?

Reply:
ok, that makes sense - and agreed to moving it up (careful with indentation - the current commit isn't what you want!), and it wouldn't make it not 1D data, but rather more than one data entry being loaded.